### PR TITLE
Medical AI - Adjust healing logic for rewrite

### PR DESCRIPTION
**Pull request by PabstMirror**
- move state machine from config to sqf
- move all healing logic to seperate func
- attempt to handle healing for new system

### CODE REVIEW

**Comment by SilentSpike** [comment]
So I gave this a very quick code review and can't see anything incorrect.

If I remember right from the meeting these changes have been playtested somewhat too. Needs to be updated for the fracture stuff, but that doesn't seem too hard. Only thing that could perhaps be improved is the way wounds are being chosen to bandage (currently just first open is used) - this might be easier once we optimise wound storage down the line though.

I reckon if fracture handling is added we should get this merged and tweak from there.

File: addons/medical_ai/functions/fnc_canRequestMedic.sqf [hunk 1]
```diff
@@ -21,10 +21,14 @@
...
 {
-    if ([_x] call EFUNC(medical_treatment,isMedic) && {!([_x] call EFUNC(common,isPlayer))}) exitWith {
-        _this setVariable [QGVAR(assignedMedic), _x];
-        true
-    };
-    false
+    if ([_x] call EFUNC(medical_treatment,isMedic) && {!([_x] call EFUNC(common,isPlayer))} && {
+                _medic = _x;
+                (unitReady _medic)
+            }) exitWith {};
```
Comment (mharis001) [inline addons/medical_ai/functions/fnc_canRequestMedic.sqf]: Formatting/indentation.

File: addons/medical_ai/functions/fnc_healingLogic.sqf [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+private _isMedic = [_healer] call EFUNC(medical_treatment,isMedic);
+private _heartRate = GET_HEART_RATE(_target);
+private _fractures = GET_FRACTURES(_target);
+
+private _treatmentEvent = "#none";
+private _treatmentArgs = [];
+private _treatmentTime = 6;
+switch (true) do {
+case (GET_WOUND_BLEEDING(_target) > 0): {
```
Comment (mharis001) [inline addons/medical_ai/functions/fnc_healingLogic.sqf]: Indentation.

File: addons/medical_ai/functions/fnc_healingLogic.sqf [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+_healer setVariable [QGVAR(currentTreatment), [CBA_missionTime + _treatmentTime, _target, _treatmentEvent, _treatmentArgs]];
+
+// Play animation
+if ((_treatmentEvent select [0,1]) != "#") then {
+    private _treatmentClassname = _treatmentArgs select 2;
+    if (_treatmentEvent == QEGVAR(medical_treatment,splintLocal)) then { _treatmentClassname = "Splint" };
+    [_healer, _treatmentClassname, (_healer == _target)] call FUNC(playTreatmentAnim);
+};
+
```
Comment (mharis001) [inline addons/medical_ai/functions/fnc_healingLogic.sqf]:
```suggestion
```

**Pull request merged**